lib/_http_outgoing.js

```diff
@@ -728,8 +728,23 @@ OutgoingMessage.prototype.setHeaders = function setHeaders(headers) {
     throw new ERR_INVALID_ARG_TYPE('headers', ['Headers', 'Map'], headers);
   }
 
-  for (const key of headers.keys()) {
-    this.setHeader(key, headers.get(key));
+  // Headers object joins multiple cookies with a comma when using
+  // the getter to retrieve the value,
+  // unless iterating over the headers directly.
+  // We also cannot safely split by comma.
+  // To avoid setHeader owerwriting the previous value we push
+  // set-cookie values in array and set them all at once.
+  const cookies = [];
+
+  for (const { 0: key, 1: value } of headers.entries()) {
```

Comment:
Isn't it better to directly loop the `headers`?

```suggestion
  for (const { 0: key, 1: value } of headers) {
```

Reply:
Headers' Symbol.iterator is equal to its entries, not sure why there'd be any performance difference.
https://github.com/nodejs/undici/blob/eb53b619fba7cde3229107d56b919e01bf2559bf/lib/fetch/headers.js#L567

Reply:
Yes, I have the same doubt when I look at the code.
But turns out it is faster in my machine, maybe anyone else can double check the performance.